pandas/core/nanops.py

```diff
@@ -521,12 +520,7 @@ def nanany(
 
     if values.dtype.kind == "M":
         # GH#34479
-        warnings.warn(
-            "'any' with datetime64 dtypes is deprecated and will raise in a "
-            "future version. Use (obj != pd.Timestamp(0)).any() instead.",
-            FutureWarning,
-            stacklevel=find_stack_level(),
-        )
+        raise TypeError("datetime64 type does not support any operations")
```

Comment:
```suggestion
        raise TypeError("datetime64 type does not support operation: 'all'")
```

Reply:
This should probably be `'any'` instead since this is for `nanany`?

Reply:
thank you, I agree, the new wording looks much better. I used the message `"does not support (any|all) operations"` because it was the original error message from `_groupby_op` and I didn't want to change it.

I corrected the messages as you suggested.